tests/run/test_genericclass.py

```diff
@@ -0,0 +1,288 @@
+### COPIED FROM CPython 3.7
+# cython: language_level=3
```

Comment:
How should I signal this test should only run for Python >= 3.7? Should I just add logic to the tests to skip based on the runtime python version?

Reply:
I'd probably drop the MRO stuff and do it in a separate full request. It's almost certainly less important for Cython since the chances are that the `typing` module will never be compiled so Cython will never see it's MRO stuff.

With that done, I think the test could actually run on any Python version.